src/main/java/com/aws/greengrass/detector/detector/IpDetector.java

```diff
@@ -44,6 +44,9 @@ List<InetAddress> getIpAddresses(Enumeration<NetworkInterface> interfaces, Confi
 
             for (InterfaceAddress interfaceAddress : networkInterface.getInterfaceAddresses()) {
                 InetAddress address = interfaceAddress.getAddress();
+                if (config.getExcludedIPAddresses().contains(address.getHostAddress())) {
```

Comment:
just a heads up this is running in a different thread than config changes, so there's technically some interleaving here when new config is cleared/set.  not a huge deal because this method is called frequently, but for completely should probably address

Reply:
and by heads up i mean, lets address it :) the worst case (although unlikely) means that connectivity info will be published with excluded ips for bit

Reply:
I'm not too concerned about this issue, because when IP Detector is deployed, the config merge happens before its startup lifecycle. When IP detector goes into startup, it should already have the deployed excluded-ip config.

We do not have any use case for its config update outside deployments as of today, since component itself does not update it and other components are not allowed to update it.

Reply:
but what about deployments that are just config updates? these exclusions wouldn't change?   even if not, there's still the mental overhead of having to justify why the race condition isn't an issue